### PR TITLE
EOS-19486: Unify deploy and deploy_vm commands

### PR DESCRIPTION
**Pull request by sradhanand-pati**
Signed-off-by: sradhanandpati <sradhanand.pati@seagate.com>

### CODE REVIEW

File: api/python/provisioner/commands/deploy.py [hunk 1]
```diff
@@ -83,22 +84,31 @@
     iopath=[
         "misc_pkgs.lustre",
         "motr",
-        "s3server"
+        "s3server",
+        "hare"
     ],
+    # The R1 HW HA sequence
```
Comment (ypise) [inline api/python/provisioner/commands/deploy.py:91]: Delete them, it creates confusion

File: api/python/provisioner/commands/deploy.py [hunk 2]
```diff
@@ -72,7 +72,8 @@
         "misc_pkgs.kafka",
         "misc_pkgs.elasticsearch",
         "misc_pkgs.kibana",
-        "misc_pkgs.statsd"
+        "misc_pkgs.statsd",
+        "misc_pkgs.consul.install"
```
Comment (ypise) [inline api/python/provisioner/commands/deploy.py]: Add **_corosync-pacemaker_** here.
Also **_ha.haproxy.start_** if necessary

File: api/python/provisioner/commands/deploy.py [hunk 3]
```diff
@@ -83,22 +84,31 @@
...
+    #     "ha.corosync-pacemaker",
+    #     "ha.haproxy.start",
+    #     "hare",
+    #     "ha.cortx-ha",
+    #     "ha.iostack-ha"
+    # ],
+
     ha=[
-        "ha.corosync-pacemaker",
+        "ha.corosync-pacemaker.install",
```
Comment (ypise) [inline api/python/provisioner/commands/deploy.py:101]: Corosync-pacemaker should be part of prereq list

File: api/python/provisioner/commands/deploy.py [hunk 4]
```diff
@@ -83,22 +84,31 @@
...
+        "ha.corosync-pacemaker.config.base",
         "ha.haproxy.start",
```
Comment (ypise) [inline api/python/provisioner/commands/deploy.py:103]: haproxy is required to be running for s3 as well. Better move this to prepreqs list

File: api/python/provisioner/commands/deploy.py [hunk 5]
```diff
@@ -83,22 +84,31 @@
...
     ],
     # states to be applied in desired sequence
     controlpath=[
         "sspl",
         "uds",
-        "csm",
-        "ha.ctrlstack-ha",
-        "ha.cortx-ha.ha"
+        "csm"
+        # "ha.ctrlstack-ha",
```
Comment (ypise) [inline api/python/provisioner/commands/deploy.py:111]: Delete the unnecessary ones

File: api/python/provisioner/commands/deploy.py [hunk 6]
```diff
@@ -268,35 +281,26 @@ def _run_states(self, states_group: str, run_args: run_args_type):
         primary = self._primary_id()
         secondaries = f"not {primary}"
 
+        hw_states = [
```
Comment (ypise) [inline api/python/provisioner/commands/deploy.py:285]: Network & firewall too comes here if we are making this special step.
Currently move this to **_system_** stage.

File: api/python/provisioner/commands/deploy.py [hunk 7]
```diff
@@ -268,35 +281,26 @@ def _run_states(self, states_group: str, run_args: run_args_type):
...
+            "misc_pkgs.ipmi.bmc_watchdog",
+        ]
+
         # apply states
-        if setup_type == SetupType.SINGLE:
-            # TODO use salt orchestration
-            for state in states:
+        for state in states:
+
+            if state in hw_states and not self._is_hw():
```
Comment (ypise) [inline api/python/provisioner/commands/deploy.py:293]: Filter HW list here from the system states.

File: api/python/provisioner/commands/deploy.py [hunk 8]
```diff
@@ -268,35 +281,26 @@ def _run_states(self, states_group: str, run_args: run_args_type):
...
+                continue
+
+            if setup_type == SetupType.SINGLE:
```
Comment (ypise) [inline api/python/provisioner/commands/deploy.py:296]: We don't need specific sync states anymore.
Also, check if we can calculate setup type based on entries in command parameters or config.ini

File: api/python/provisioner/commands/deploy.py [hunk 9]
```diff
@@ -268,35 +281,26 @@ def _run_states(self, states_group: str, run_args: run_args_type):
...
-                            stages
-                        )
-
-                elif state in (
+            else:
+                if state in (
                     "system.storage",
                     "sspl",
-                    "csm",
+                    "ha.cortx-ha",
```
Comment (ypise) [inline api/python/provisioner/commands/deploy.py:304]: For **_cortx-ha_**, it should be sequentially on all nodes.
Check with @83bhp

File: api/python/provisioner/commands/deploy.py [hunk 10]
```diff
@@ -324,48 +340,41 @@ def _update_salt(self, targets=config.ALL_MINIONS):
...
     def _destroy_storage(self, run_args, nofail=True):
         targets = run_args.targets
 
-        # FIXME VERIFY EOS-12076
-        if (
-            (run_args.setup_type != SetupType.SINGLE) and
-            (targets == config.ALL_MINIONS)
-        ):
-            targets = f"not {self._primary_id()}"
+        # # FIXME VERIFY EOS-12076
```
Comment (ypise) [inline api/python/provisioner/commands/deploy.py:363]: Delete if not required.

File: api/python/provisioner/commands/deploy_vm.py [hunk 11]
```diff
@@ -102,50 +102,6 @@ class DeployVM(Deploy):
     _run_args_type = run_args_type
     setup_ctx: Optional[SetupCtx] = None
 
-    def set_pillar_data(self, setup_type, targets):
```
Comment (ypise) [inline api/python/provisioner/commands/deploy_vm.py:105]: Why not delete the file?

**Review by ypise — CHANGES_REQUESTED** [review]

File: api/python/provisioner/commands/deploy_vm.py (hunk 11 above)
Comment (sradhanand-pati) [reply api/python/provisioner/commands/deploy_vm.py:105]: will delete it once it passes multiple rounds of testing.

**Review by ypise — APPROVED** [review]

**Pull request merged**